### ISSUE TITLE
Segfault with Bitvector Arrays

### ISSUE THREAD

**Issue by evanlohn** [bitwuzla/bitwuzla#50]
In commit [3130e49](https://github.com/bitwuzla/bitwuzla/commit/1230d80a5275ddf525da9c218c2bcd29c3130e49) on Amazon Linux compiled with gcc-7.3.1, declaring an array and then defining a function with a quantifier results in a segfault, even when the function does not use the array. Example below:

```
(declare-const s (Array (_ BitVec 8) (_ BitVec 8)))
(define-fun f 
  ((a (_ BitVec 8)))
  Bool
  (forall ((j (_ BitVec 8))) (bvult (_ bv0 8) a)))
  
(assert (f (_ bv0 8)))
(check-sat)
```

In debug mode, this gives

```
Assertion failed: (free_vars.size() > 0), function compute_variable_dependencies, file bzlaslvquant.cpp, line 2052.
[bitwuzla>main] CAUGHT SIGNAL 6
unknown
Abort
```

Removing the forall or the declaration of s gets rid of the segfault. I have also run into segfaults using Array constants in function bodies, but I'm hoping fixing this will fix that problem as well since the same assertion fails in debug mode.

**Comment by mpreiner** [comment]
Hi @evanlohn, thanks for the bug report. The new from-scratch rewrite of Bitwuzla does not have this issue anymore. Can you also post the example with the array constant in the function bodies?

**Comment by evanlohn** [comment]
I can't post the actual example due to confidentiality, but this modification also segfaults and has the array in the function body:

```(declare-const s (Array (_ BitVec 8) (_ BitVec 8)))
(define-fun f 
  ((a (_ BitVec 8)))
  Bool
  (forall ((j (_ BitVec 8))) (and (= s s) (bvult (_ bv0 8) a))))

(assert (f (_ bv0 8)))
(check-sat)
```

**Comment by mpreiner** [comment]
Fixed in new 0.1.0 release.
